src/components/Articles.jsx

```diff
@@ -2,22 +2,74 @@ import React, { useEffect, useState } from "react";
 import { getArticles } from "../utils";
 import { v4 as uuidv4 } from "uuid";
 import ArticlesCard from "./ArticlesCard";
-import { useParams } from "react-router-dom";
+
+import { useParams, useSearchParams } from "react-router-dom";
 import "../styles/Articles.css";
 
 function Articles() {
   const [articles, setArticles] = useState([]);
+  const [searchParams, setSearchParams] = useSearchParams();
+  const [sortBy, setSortBy] = useState("created_at");
+  const [order, setOrder] = useState("asc");
   const { topic } = useParams();
-
   useEffect(() => {
-    getArticles(topic).then((articlesFromApi) => {
+    getArticles(topic, searchParams).then((articlesFromApi) => {
       setArticles(articlesFromApi);
     });
-  }, [topic]);
+  }, [topic, searchParams]);
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+
+    let params = {
+      sort_by: sortBy,
+      order: order,
+    };
+
+    const newSearchParams = new URLSearchParams(params);
+    return setSearchParams(newSearchParams);
+  };
 
   return (
     <section>
```

Comment:
This is probably enough functionality to be a separate component

Reply:
I agree, hesitated on it last minute. Refactor list added.